osgp-adapter-ws-shared/src/main/java/com/alliander/osgp/adapter/ws/shared/services/AbstractResendNotificationService.java

```
@@ -65,7 +66,14 @@ public void execute() {
              * the results this eliminates the chance of repeatedly getting a
              * full page of response data of which none are actually re-sent.)
              */
-            this.resendNotifications((short) notificationsResent, createdBefore);
+            try {
+                this.resendNotifications((short) notificationsResent, createdBefore);
+            } catch (final CircuitBreakerOpenException exc) {
```

I think this catch of the `CircuitBreakerOpenException` looks like it belongs with the (initial) `NotificationService`'s `sendNotification` as well. Now `AbstractNotificationServiceWs.doSendNotification` catches only a number of other exceptions. Or is it supposed to be different somehow (send vs resend, and the other caught exceptions vs the uncaught `CircuitBreakerOpenException`)?

---

I put it in this location to end the loop the `AbstractResendNotificationService.execute()` method in a relatively clean way. To do so, I want to propagate the exception to this level. The `execute()` method calls `NotificationServiceWs.sendNotification()` for each notification, therefore I can't catch it yet in a class implementing `NotificationService.sendNotification()`.

`AbstractNotificationServiceWs.sendNotification()` catches "WebServiceSecurityException | WebServiceIOException | SoapFaultClientException". It's not clear to me why this specific set of exceptions is caught. @smvdheijden didn't know either where that set came from.

---

My worries are that the `CircuitBreakerOpenException` will lead to inappropriate handling of the notification when it is sent initially (and thus not handled the way it is with `resendNotifications`.
If handling of the exception happens OK in either scenario, the code is fine with me. If the application may behave inappropriate in some cases with the code as it is, I think some other (maybe less clean if unavoidable) way of processing notifications is in order.